src/daft-core/src/array/ops/hash.rs

```diff
@@ -23,73 +27,121 @@ impl<T> DataArray<T>
 where
     T: DaftPrimitiveType,
 {
-    pub fn hash(&self, seed: Option<&UInt64Array>) -> DaftResult<UInt64Array> {
+    pub fn hash(
+        &self,
+        seed: Option<&UInt64Array>,
+        hash_function: HashFunctionKind,
+    ) -> DaftResult<UInt64Array> {
+        self.hash_with_specified_algorithm(seed, hash_function)
+    }
+    pub fn hash_with_specified_algorithm(
+        &self,
+        seed: Option<&UInt64Array>,
+        _hash_function: HashFunctionKind,
+    ) -> DaftResult<UInt64Array> {
         let as_arrowed = self.as_arrow();
-
         let seed = seed.map(|v| v.as_arrow());
-
-        let result = kernels::hashing::hash(as_arrowed, seed)?;
-
+        let result = kernels::hashing::hash(as_arrowed, seed, _hash_function)?;
         Ok(DataArray::from((self.name(), Box::new(result))))
     }
 }
 
 impl Utf8Array {
-    pub fn hash(&self, seed: Option<&UInt64Array>) -> DaftResult<UInt64Array> {
+    pub fn hash(
+        &self,
+        seed: Option<&UInt64Array>,
+        hash_function: HashFunctionKind,
+    ) -> DaftResult<UInt64Array> {
+        self.hash_with_specified_algorithm(seed, hash_function)
+    }
+    pub fn hash_with_specified_algorithm(
+        &self,
+        seed: Option<&UInt64Array>,
+        _hash_function: HashFunctionKind,
+    ) -> DaftResult<UInt64Array> {
         let as_arrowed = self.as_arrow();
-
         let seed = seed.map(|v| v.as_arrow());
-
-        let result = kernels::hashing::hash(as_arrowed, seed)?;
-
+        let result = kernels::hashing::hash(as_arrowed, seed, _hash_function)?;
         Ok(DataArray::from((self.name(), Box::new(result))))
     }
 }
 
 impl BinaryArray {
-    pub fn hash(&self, seed: Option<&UInt64Array>) -> DaftResult<UInt64Array> {
+    pub fn hash(
+        &self,
+        seed: Option<&UInt64Array>,
+        hash_function: HashFunctionKind,
+    ) -> DaftResult<UInt64Array> {
+        self.hash_with_specified_algorithm(seed, hash_function)
+    }
```

Comment:
Same feedback here. This should significantly reduce the diff because you won't have to update all callers of `.hash()` on all of the array types